xxh3.h

```diff
@@ -155,9 +155,132 @@
 #   warning "XXH3 is highly inefficient without ARM or Thumb-2."
 #endif
 
+/* ==========================================
+ * Dispatcher macros
```

Comment:
One annoyance in this PR is that there is a significant amount of code for the dispatcher inside `xxh3.h`.
Not that I'm expecting `xxh3.h` to remain unmodified, it's clear that it is necessary to add a few hooks,
but I was wondering if changes within `xxh3.h` could be reduced, and rather transferred into `xxhash-dispatch.c`.
For example, they could be defined within `xxhash-dispatch.c` _before_ including `xxhash.h`

Reply:
Well we could have an `ifndef/define` guard and put everything dispatch in another file, but that option causes issues with most `XXH_INLINE_ALL` uses.

If you are fine with that, I'm fine with that.

Alternatively, we could have `xxhash.h` include a different header if `XXH_DISPATCH` is defined, but that makes things complicated with build paths and can be ugly when it is installed as a system header.

Reply:
> that option causes issues with most `XXH_INLINE_ALL` uses.

Well, the objective being to reduce potential impact on normal behavior of `xxhash.h`, 
this outcome wouldn't be fine.

That being said, I'm struggling to understand a scenario where inlining would be impacted.

Reply:
Unless you include the special header beforehand, the preinclude method would only use the normal code path with XXH_INLINE_ALL.

The include from xxhash.h adds *another* header to `/usr/include`, which is unwanted.

Reply:
> Unless you include the special header beforehand,

Well yes that's the idea

Reply:
I mean, we could nop the header on other arches.

So basically it could be like this
```c
// xxhash-dispatch.h
#if defined(__i386__) || defined(__x86_64__) || ...

#define XXH_DISPATCH_CALL(...) ...
#define ...
...
#endif
#include "xxhash.h"
```

Reply:
Yes, exactly